pkg/apis/core/v1/helper/helpers.go

```diff
@@ -436,3 +438,15 @@ func StorageNodeAffinityToAlphaAnnotation(annotations map[string]string, affinit
 	annotations[v1.AlphaStorageNodeAffinityAnnotation] = string(json)
 	return nil
 }
+
+// IsCriticalPodBasedOnPriority checks if the given pod is a critical pod based on priorityClass mentioned in pod Spec.
+func IsCriticalPodBasedOnPriority(ns string, priorityClassName string) bool {
+	// Critical pods are restricted to "kube-system" namespace as of now.
+	if ns != kubeapi.NamespaceSystem {
+		return false
+	}
+	if _, ok := schedulerapi.SystemPriorityClasses[priorityClassName]; ok {
```

Comment:
In the design of Priority, we decided that internal components of K8s shouldn't care about the `PriorityClassName` and should always look at the value of priority. So, please change the function to receive the integer value of priority (PodSpec.Priority) and change this condition to:
```
if podPriority >= schedulerapi.SystemCriticalPriority {
```

Reply:
Done.